### PR TITLE
px-monitor helm chart

### PR DESCRIPTION
**Pull request by tusharraut1994**
Signed-off-by: Tushar Raut tushar@portworx.com

- Added cortex stack templates
- Added px-monitor post-install setup and pre-delete jobs
- Added Grafana and Prometheus specs 
- Updated the frontend proxy configuration to redirect grafana and cortex endpoints
- Added README for px-monitor chart
- Added Prometheus crds 
- Added portworx and px-backup dashboard files

### CODE REVIEW

File: charts/px-monitor/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (sharma-tapas) [inline charts/px-monitor/Chart.yaml:1]: Add a maintainers section too.

File: charts/px-monitor/templates/cortex/cortex-backend.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  {{- if eq .Values.persistentStorage.enabled true }}
+  volumeClaimTemplates:
+    - metadata:
+        name: cassandra-data
+      spec:
+        accessModes: ["ReadWriteOnce"]
+        storageClassName: {{ .Values.persistentStorage.storageClassName }}
+        resources:
+          requests:
+            storage: 50Gi
```
Comment (sharma-tapas) [inline charts/px-monitor/templates/cortex/cortex-backend.yaml:111]: Will this be enough or should we make it configurable via the values.yaml? Let the default be 50G and user can override if needed?

File: charts/px-monitor/templates/cortex/cortex-proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+apiVersion: v1
+metadata:
+  name: pxcentral-cortex-nginx
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: cortex-nginx
+    cortex: nginx
+{{- include "px-monitor.labels" . | nindent 4 }}
+data:
+  nginx.conf: |-
```
Comment (sharma-tapas) [inline charts/px-monitor/templates/cortex/cortex-proxy.yaml:14]: Can we read this from a file and use https://helm.sh/docs/chart_template_guide/accessing_files/

File: charts/px-monitor/templates/grafana/grafana-config.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+kind: ConfigMap
+metadata:
+  name: pxcentral-grafana-ini-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    grafana: portworx
+    app.kubernetes.io/component: pxcentral-grafana
+{{- include "px-monitor.labels" . | nindent 4 }}
+data:
+  grafana.ini: |
```
Comment (sharma-tapas) [inline charts/px-monitor/templates/grafana/grafana-config.yaml:15]: check https://helm.sh/docs/chart_template_guide/accessing_files/

File: charts/px-monitor/templates/cortex/cortex-backend.yaml (hunk 2 above)
Comment (tusharraut1994) [reply charts/px-monitor/templates/cortex/cortex-backend.yaml:111]: Added

File: charts/px-monitor/Chart.yaml (hunk 1 above)
Comment (tusharraut1994) [reply charts/px-monitor/Chart.yaml:1]: Added

File: charts/px-monitor/templates/cortex/cortex-proxy.yaml (hunk 3 above)
Comment (tusharraut1994) [reply charts/px-monitor/templates/cortex/cortex-proxy.yaml:14]: This is not a static file and needs to updated based on inputs. so can't add and load from files directory.

File: charts/px-monitor/templates/grafana/grafana-config.yaml (hunk 4 above)
Comment (tusharraut1994) [reply charts/px-monitor/templates/grafana/grafana-config.yaml:15]: This is not a static file and needs to updated based on inputs. so can't add and load from files directory

**Pull request merged**